src/pudl/package_data/meta/datapkg/datapackage.json

```diff
@@ -5475,6 +5475,83 @@
                 "path": "https://creativecommons.org/licenses/by/4.0/"
             }],
             "sources": []
+        },
+        {
+            "profile": "tabular-data-resource",
+            "name": "assn_eia_epacems",
+            "path": "data/assn_eia_epacems.csv",
+            "title": "assn_eia_epacems",
+            "encoding": "utf-8",
+            "mediatype": "text/csv",
+            "format": "csv",
+            "dialect": {
+                "delimiter": ",",
+                "header": true,
+                "quoteChar": "\"",
+                "doubleQuote": true,
+                "lineTerminator": "\r\n",
+                "skipInitialSpace": true,
+                "caseSensitiveHeader": false
+            },
+            "schema": {
+                "fields": [{
+                        "name": "plant_id_eia",
+                        "type": "integer",
+                        "format": "default",
+                        "description": "The unique six-digit facility identification number, also called an ORISPL, assigned by the Energy Information Administration."
+                    },
+                    {
+                        "name": "plant_id_epa",
+                        "type": "integer",
+                        "format": "default"
+                    },
+                    {
+                        "name": "plant_name_eia",
+                        "type": "string",
+                        "format": "default"
+                    },
+                    {
+                        "name": "epa_point_source_unit",
+                        "type": "integer",
+                        "format": "default",
+                        "description": "'Smokestack' unit monitored by EPA CEMS."
+                    },
+                    {
+                        "name": "generator_id",
+                        "type": "integer",
+                        "format": "default"
+                    },
+                    {
+                        "name": "boiler_id",
+                        "type": "integer",
+                        "format": "default"
+                    },
```

Comment:
It looks like a lot of these IDs are actually strings, rather than integers, so the datapackage validation is currently failing.  The type (and potentially format) need to be compatible with whatever the contents of the column are in the CSV.

Reply:
what exactly does the "format" mean?

Reply:
A CSV contains characters which make up strings, but the various columns of the CSV are meant to be interpreted as things that aren't necessarily strings. The `format` attribute indicates how the string stored in the CSV should be interpreted as... whatever type it's supposed to ultimately turn into (e.g. there's lots of different ways to encode a date or a time as a string). Lots of them are simple and only have the default case, but some of them are more complicated.

All of these attributes are defined by the [TableSchema spec](https://specs.frictionlessdata.io/table-schema/), as are the various ways of defining primary keys, foreign key constraints, values that should be interpreted as True or False or NA, etc.